internal/v2/controller/http/callback.go

```diff
@@ -0,0 +1,189 @@
+	}
+	if err == nil {
+		c.lc.Debug(fmt.Sprintf("Invoked driver.AddDevice/UpdateDevice callback for %s", device.Name))
+	} else {
+		errMsg := fmt.Sprintf("driver.AddDevice/UpdateDevice callback failed for %s", device.Name)
+		c.sendError(writer, request, edgexErr.KindServerError, errMsg, err, sdkCommon.APIV2DeviceCallbackRoute, correlationID)
+		return
+	}
+
+	c.lc.Debug(fmt.Sprintf("Handler - starting AutoEvents for device %s", device.Name))
+	autoevent.GetManager().RestartForDevice(device.Name, dic)
+
+	c.sendResponse(writer, request, sdkCommon.APIV2DeviceCallbackRoute, nil, http.StatusOK)
+}
+
+func (c *V2HttpController) DeleteProfile(writer http.ResponseWriter, request *http.Request, _ *di.Container) {
+	vars := mux.Vars(request)
+	id := vars[sdkCommon.IdVar]
+	correlationID := request.Header.Get(sdkCommon.V2CorrelationHeaderKey)
+
+	err := cache.Profiles().Remove(id)
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to remove profile with given id %s", id)
+		c.sendError(writer, request, edgexErr.KindInvalidId, errMsg, err, sdkCommon.APIV2ProfileCallbackIdRoute, correlationID)
+		return
+	}
+
+	c.lc.Info(fmt.Sprintf("Removed profile with given id %s", id))
+	c.sendResponse(writer, request, sdkCommon.APIV2ProfileCallbackIdRoute, nil, http.StatusOK)
+}
+
+func (c *V2HttpController) AddUpdateProfile(writer http.ResponseWriter, request *http.Request, _ *di.Container) {
+	method := request.Method
+	correlationID := request.Header.Get(sdkCommon.V2CorrelationHeaderKey)
+
+	// TODO: change to v2dto when core services are ready.
+	var profile contract.DeviceProfile
+	err := json.NewDecoder(request.Body).Decode(&profile)
+	if err != nil {
+		c.sendError(writer, request, edgexErr.KindContractInvalid, "JSON decode failed", err, sdkCommon.APIV2ProfileCallbackRoute, correlationID)
+		return
+	}
+
+	if method == http.MethodPost {
+		err = cache.Profiles().Add(profile)
+	} else {
+		err = cache.Profiles().Update(profile)
+	}
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to %s profile %s", method, profile.Name)
+		c.sendError(writer, request, edgexErr.KindServerError, errMsg, err, sdkCommon.APIV2ProfileCallbackRoute, correlationID)
+		return
+	}
+
+	c.lc.Info(fmt.Sprintf("profile %s added/updated", profile.Name))
+	c.sendResponse(writer, request, sdkCommon.APIV2ProfileCallbackRoute, nil, http.StatusOK)
+}
+
+func (c *V2HttpController) DeleteWatcher(writer http.ResponseWriter, request *http.Request, _ *di.Container) {
+	vars := mux.Vars(request)
+	id := vars[sdkCommon.IdVar]
+	correlationID := request.Header.Get(sdkCommon.V2CorrelationHeaderKey)
+
+	err := cache.ProvisionWatchers().Remove(id)
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to remove provisionwatcher with given id %s", id)
+		c.sendError(writer, request, edgexErr.KindInvalidId, errMsg, err, sdkCommon.APIV2WatcherCallbackIdRoute, correlationID)
+		return
+	}
+
+	c.lc.Info(fmt.Sprintf("Removed provisionwatcher with given id %s", id))
```

Comment:
This log message should be moved into Remove() which services as the application layer for this API.